library/extractor/src/main/java/com/google/android/exoplayer2/text/subrip/SubripDecoder.java

```diff
@@ -75,8 +77,25 @@ protected Subtitle decode(byte[] bytes, int length, boolean reset) {
     LongArray cueTimesUs = new LongArray();
     ParsableByteArray subripData = new ParsableByteArray(bytes, length);
 
+    @Nullable Charset utf16Charset;
+    if (bytes.length >= 2) {
+      utf16Charset = getUtf16Charset(bytes[0], bytes[1]);
+    } else {
+      utf16Charset = null;
+    }
+
     @Nullable String currentLine;
-    while ((currentLine = subripData.readLine()) != null) {
+    while (true) {
+      if (utf16Charset != null) {
+        currentLine = subripData.readLineUtf16(utf16Charset.equals(Charsets.UTF_16LE));
+      } else {
+        currentLine = subripData.readLine();
```

Comment:
I don't think that is good solution and I did it on purpose.

If you have a method that contains Charset in signature. The calling site will assume that any Charset can be passed and it will work well. 
But in this case, we will only support 2 encodings from the entire set that contains Charsets class.
So, in my opinion, I don't think this is a good solution. What about you?

Reply:
Ok, resolved by creating `ParsableByteArray.readUtfLine(Charset)`